### PR TITLE
Add scope hard coded checks in places employee permissions were hard coded

### LINKED ISSUE #35670: Some core services have internal manual check of permission, this is not compatible with the API at all
## Issue

Example:

Module actions that depend on ModuleManager and more specifically https://github.com/PrestaShop/PrestaShop/blob/efb6aa7a6492d6533ecdbaa4a9477135a52eb973/src/Adapter/Module/AdminModuleDataProvider.php#L170

The check is only performed on a logged in employee, we need to allow checking access based on an API client as well, this implies thinking about what scope we'd use, do we introduce core scopes manually? Do we think of a way to convert legacy permissions into scopes?
We need to search in the core what other part are similar to this and could be blocked so we anticipate the solution before v9 release.

Employe::can

Image

Also check `Profile::getProfileAccesses` and `Profile::getProfileAccess`

## Investigation 
- List endpoints impacted (17 endpoints)
- Can we find a generic solution ? POC

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add scope hard-coded checks in places employee permissions were hard coded, this is not a perfect solution but a quick one The module and theme endpoints will simply have to match the chosen scope when they are implemented Only one write scope is checked, any more advanced scope policy will require a new development or a refacto to handle this more generically
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/8438552754
| UI Tests          | Fixes #35670
| Fixed issue or discussion?     | ~
| Related PRs       | https://github.com/PrestaShop/ps_apiresources/pull/20
| Sponsor company   | ~

### Breaking changes

- `PrestaShop\PrestaShop\Core\Addon\Module\ModuleManagerBuilder` public static fields were changed into protected ones as they were not used outside of the class
- `PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManager` internal fields were turned into readonly fields

### CODE REVIEW

File: src/Adapter/Module/Repository/ModuleRepository.php [hunk 1]
```diff
@@ -29,6 +29,7 @@
 namespace PrestaShop\PrestaShop\Adapter\Module\Repository;
 
 use Exception;
+use Module;
```
Comment (tleon) [inline src/Adapter/Module/Repository/ModuleRepository.php]: How can this be the only thing modified in this file ? didn't you forget to remove some `\`?

**Review by tleon — DISMISSED** [review]

File: src/Core/Addon/Module/ModuleManagerBuilder.php [hunk 2]
```diff
@@ -56,13 +57,14 @@ class ModuleManagerBuilder
...
      * @var ModuleManager
      */
-    public static $moduleManager = null;
-    public static $adminModuleDataProvider = null;
-    public static $legacyLogger = null;
-    public static $moduleDataProvider = null;
-    public static $translator = null;
-    public static $instance = null;
-    public static $cacheProvider = null;
+    protected static $moduleManager = null;
```
Comment (matthieu-rolland) [inline src/Core/Addon/Module/ModuleManagerBuilder.php:60]: shouldn't this be noted as a BC break?

File: src/Adapter/Module/Repository/ModuleRepository.php (hunk 1 above)
Comment (jolelievre) [reply src/Adapter/Module/Repository/ModuleRepository.php]: Ah it's the remnant of a previous implementation I need to clean this indeed

**Review by matthieu-rolland — DISMISSED** [review]
One small feedback which doesn't require changes in the codebase, LGTM otherwise 👍

File: src/Core/Addon/Module/ModuleManagerBuilder.php (hunk 2 above)
Comment (jolelievre) [reply src/Core/Addon/Module/ModuleManagerBuilder.php:60]: Yes you're right

**Review by matthieu-rolland — APPROVED** [review]

**Review by tleon — APPROVED** [review]

**Pull request merged**